### PR TITLE
enabled video request on client side, added parameters to set video constraints

### PR DESCRIPTION
**Pull request by zemeng**
changes made to allow users make video stream request.  users are also able to set video constraints.

### CODE REVIEW

File: src/js/rtc_session.js [hunk 1]
```diff
@@ -447,6 +447,13 @@ export default class RtcSession {
         } else {
             this._callId = contactId;
         }
+
+        //logging
+        console.log("signalingUri: " + JSON.stringify(signalingUri));
```
Comment (yuliangz) [inline src/js/rtc_session.js]: please remove debug logging

File: src/js/rtc_session.js [hunk 2]
```diff
@@ -598,6 +609,16 @@ export default class RtcSession {
     set minVideoHeight(height) {
         this._minVideoHeight = height;
     }
+    set videoHeight(height) {
```
Comment (yuliangz) [inline src/js/rtc_session.js]: name it explicitly "idealVideoHeight"?

File: src/js/rtc_session.js [hunk 3]
```diff
@@ -839,13 +863,62 @@ export default class RtcSession {
...
+            }
+            if(Object.keys(frameRateConstraints).length > 0) {
+                videoConstraints.frameRate = frameRateConstraints;
+            }
+
+            // build facing mode constraints
+            if(self._facingMode === 'user' || self._facingMode === "environment") {
+                videoConstraints.facingMode = self._facingMode;
+            } else {
+                self._facingMode = 'user';
```
Comment (yuliangz) [inline src/js/rtc_session.js]: should set videoContraints here?

File: src/js/rtc_session.js [hunk 4]
```diff
@@ -839,13 +863,62 @@ export default class RtcSession {
...
+            }
+
+            // set video constraints
             if (Object.keys(videoConstraints).length > 0) {
                 mediaConstraints.video = videoConstraints;
             } else {
                 mediaConstraints.video = true;
             }
-            //TODO facingMode : 'user' or 'environment'
+            console.log("Video Constraints: " + JSON.stringify(videoConstraints));
```
Comment (yuliangz) [inline src/js/rtc_session.js]: remove debug log

File: demovideo/index.html [hunk 5]
```diff
@@ -0,0 +1,29 @@
+<html>
+
+<head>
+    <title>connect-rtc-js demo video</title>
```
Comment (yuliangz) [inline demovideo/index.html]: Could you merge the logic into the existing demo page by adding media options?

File: src/js/rtc_session.js (hunk 1 above)
Comment (zemeng) [inline src/js/rtc_session.js]: fixed. debug logging were removed in the new iteration.

File: demovideo/index.html (hunk 5 above)
Comment (zemeng) [inline demovideo/index.html]: updated. video demo page was removed in the new iteration, audio & video logic are now merged into the same demo page.

File: src/js/rtc_session.js (hunk 2 above)
Comment (zemeng) [inline src/js/rtc_session.js]: fixed.

File: src/js/rtc_session.js (hunk 3 above)
Comment (zemeng) [inline src/js/rtc_session.js]: fixed.

File: src/js/rtc_session.js (hunk 4 above)
Comment (zemeng) [inline src/js/rtc_session.js]: fixed. it's removed in the new iteration.

**Pull request merged**